### ISSUE TITLE
java-quarkus Devfile stack fails to build `community` starter project

### ISSUE THREAD

**Issue by johnmcollier** [quarkusio/quarkus#23562]
### Describe the bug

I downloaded the `java-quarkus` stack from the [Devfile Registry](https://registry.devfile.io/viewer/devfiles/Community+java-quarkus), and used [odo](https://github.com/redhat-developer/odo) to deploy it onto Kubernetes, using the `community` starter project (https://code.quarkus.io/d?e=io.quarkus%3Aquarkus-resteasy&e=io.quarkus%3Aquarkus-micrometer&e=io.quarkus%3Aquarkus-smallrye-health&e=io.quarkus%3Aquarkus-openshift&cn=devfile
). But the application never becomes available and I see the following in the container logs:

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.827 s
[INFO] Finished at: 2022-02-09T18:12:39Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.7.1.Final:dev (default-cli) on project code-with-quarkus: Detected Maven Version (3.6.2)  is not supported, it must be in [3.8.1,). -> [Help 1]
```

CC @maxandersen 

### Expected behavior

The `community` starter project is built successfully and becomes available.

### Actual behavior

The `community` starter project fails to build.

### How to Reproduce?

1. Download [odo](github.com/redhat-developer/odo)
2. Use odo to download and deploy the `java-quarkus` devfile stack on OpenShift
  a. `odo create java-quarkus --starter community`
  b. `odo push`
3. The command will "succeed" but if you attempt to access the exposed application, the application will not be running. If you look at the container logs for Quarkus, you should see the following error:
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.827 s
[INFO] Finished at: 2022-02-09T18:12:39Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.7.1.Final:dev (default-cli) on project code-with-quarkus: Detected Maven Version (3.6.2)  is not supported, it must be in [3.8.1,). -> [Help 1]
```

### Additional information

We first started detecting this failure on the Devfile Registry CI on February 2nd: https://github.com/devfile/registry/actions/runs/1785051164, with the previous test run on Jan 24th passing: https://github.com/devfile/registry/actions/runs/1741799549

(Unfortunately we don't have the container logs from those test runs, so it may not help with debugging).

**Comment by johnmcollier** [comment]
Spoke with @maxandersen and he asked me to open an issue here for this issue

**Comment by geoand** [comment]
So this is about the maven version being out of date?
Where is it defined?

**Comment by johnmcollier** [comment]
@geoand It's coming from the registry.access.redhat.com/ubi8/openjdk-11:latest container image:
```
johncollier@jcollier-mac ~ % docker run registry.access.redhat.com/ubi8/openjdk-11:latest mvn --version 
Apache Maven 3.6.2 (Red Hat 3.6.2-6)
Maven home: /usr/share/maven
Java version: 11.0.14, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-11-openjdk-11.0.14.0.9-2.el8_5.x86_64
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "5.10.47-linuxkit", arch: "amd64", family: "unix"
```

Which is specified here in the [java-quarkus devfile](https://github.com/devfile/registry/blob/main/stacks/java-quarkus/devfile.yaml#L22).

**Comment by geoand** [comment]
Thanks.

@maxandersen not sure what we can do here...

**Comment by maxandersen** [comment]
we'll need to either get mvn 3.8 included in ubi8 or we need to investigate if we can't avoid having to call out to 3.8 only API unless absolutely necessary.

@aloubyansky is it really required ?

**Comment by maxandersen** [comment]
like, it seem to build and run just fine with 2.6.x - no registry access needed afaics.

**Comment by elsony** [comment]
Do we happen to have an update on this?

**Comment by maxandersen** [comment]
it should be fixed with latest 2.7.2 quarkus so it doesn't fail. 

It was released last night so if you run the tests again you should see it pass.

**Comment by johnmcollier** [comment]
No longer seeing the failure, so closing this out.

Thanks for fixing this 👍
